src/python_testing/TestConformanceTest.py

```diff
@@ -0,0 +1,131 @@
+                                        if is_mandatory(a.conformance)]
+        attrs[GlobalAttributeIds.GENERATED_COMMAND_LIST_ID] = mandatory_generated_commands
+        attrs[GlobalAttributeIds.CLUSTER_REVISION_ID] = self.xml_clusters[cluster_id].revision
+        return attrs
+
+    def _create_minimal_dt(self, device_type_id: int) -> dict[int, dict[int, Any]]:
+        ''' Creates the internals of an endpoint_tlv with the minimal set of clusters, with the minimal set of attributes and commands. Global attributes only.
+            Does NOT take into account overrides yet.
+        '''
+        endpoint_tlv = {}
+        required_servers = [id for id, c in self.xml_device_types[device_type_id].server_clusters.items()
+                            if is_mandatory(c.conformance)]
+        required_clients = [id for id, c in self.xml_device_types[device_type_id].client_clusters.items()
+                            if is_mandatory(c.conformance)]
+        device_type_revision = self.xml_device_types[device_type_id].revision
+
+        for s in required_servers:
+            endpoint_tlv[s] = self._create_minimal_cluster(s)
+
+        # Descriptor
+        attr = Clusters.Descriptor.Attributes
+        attrs = {}
+        attrs[attr.FeatureMap.attribute_id] = 0
+        attrs[attr.AcceptedCommandList.attribute_id] = []
+        attrs[attr.GeneratedCommandList.attribute_id] = []
+        attrs[attr.ClusterRevision.attribute_id] = self.xml_clusters[Clusters.Descriptor.id].revision
+        attrs[attr.DeviceTypeList.attribute_id] = [
+            Clusters.Descriptor.Structs.DeviceTypeStruct(deviceType=device_type_id, revision=device_type_revision)]
+        attrs[attr.ServerList.attribute_id] = required_servers
+        attrs[attr.ClientList.attribute_id] = required_clients
+        attrs[attr.PartsList.attribute_id] = []
+        attrs[attr.AttributeList.attribute_id] = []
+        attrs[attr.AttributeList.attribute_id] = list(attrs.keys())
+
+        endpoint_tlv[Clusters.Descriptor.id] = attrs
+        return endpoint_tlv
+
+    def add_macl(self, root_endpoint: dict[int, dict[int, Any]]):
+        ac = Clusters.AccessControl
+        root_endpoint[ac.id][ac.Attributes.FeatureMap.attribute_id] = ac.Bitmaps.Feature.kManagedDevice
+        root_endpoint[ac.id][ac.Attributes.Arl.attribute_id] = []
+        root_endpoint[ac.id][ac.Attributes.CommissioningARL.attribute_id] = []
+        root_endpoint[ac.id][ac.Attributes.AttributeList.attribute_id].extend([
+            ac.Attributes.Arl.attribute_id, ac.Attributes.CommissioningARL.attribute_id])
+        root_endpoint[ac.id][ac.Attributes.AcceptedCommandList.attribute_id].append(ac.Commands.ReviewFabricRestrictions.command_id)
+        root_endpoint[ac.id][ac.Attributes.GeneratedCommandList.attribute_id].append(
+            ac.Commands.ReviewFabricRestrictionsResponse.command_id)
+
+    @ async_test_body
```

Comment:
It's odd that there is a space here between the `@` and `async_test_body`

Reply:
yeah, that is odd. Fixed.